<?xml version="1.0" encoding="utf-8" ?>
<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
<html xmlns="http://www.w3.org/1999/xhtml" xml:lang="en" lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="Docutils 0.12: http://docutils.sourceforge.net/" />
<title>HeatLogo</title>
<style type="text/css">

/*
:Author: David Goodger (goodger@python.org)
:Id: $Id: html4css1.css 7614 2013-02-21 15:55:51Z milde $
:Copyright: This stylesheet has been placed in the public domain.

Default cascading style sheet for the HTML output of Docutils.

See http://docutils.sf.net/docs/howto/html-stylesheets.html for how to
customize this style sheet.
*/

/* used to remove borders from tables and images */
.borderless, table.borderless td, table.borderless th {
  border: 0 }

table.borderless td, table.borderless th {
  /* Override padding for "table.docutils td" with "! important".
     The right padding separates the table cells. */
  padding: 0 0.5em 0 0 ! important }

.first {
  /* Override more specific margin styles with "! important". */
  margin-top: 0 ! important }

.last, .with-subtitle {
  margin-bottom: 0 ! important }

.hidden {
  display: none }

a.toc-backref {
  text-decoration: none ;
  color: black }

blockquote.epigraph {
  margin: 2em 5em ; }

dl.docutils dd {
  margin-bottom: 0.5em }

object[type="image/svg+xml"], object[type="application/x-shockwave-flash"] {
  overflow: hidden;
}

/* Uncomment (and remove this text!) to get bold-faced definition list terms
dl.docutils dt {
  font-weight: bold }
*/

div.abstract {
  margin: 2em 5em }

div.abstract p.topic-title {
  font-weight: bold ;
  text-align: center }

div.admonition, div.attention, div.caution, div.danger, div.error,
div.hint, div.important, div.note, div.tip, div.warning {
  margin: 2em ;
  border: medium outset ;
  padding: 1em }

div.admonition p.admonition-title, div.hint p.admonition-title,
div.important p.admonition-title, div.note p.admonition-title,
div.tip p.admonition-title {
  font-weight: bold ;
  font-family: sans-serif }

div.attention p.admonition-title, div.caution p.admonition-title,
div.danger p.admonition-title, div.error p.admonition-title,
div.warning p.admonition-title, .code .error {
  color: red ;
  font-weight: bold ;
  font-family: sans-serif }

/* Uncomment (and remove this text!) to get reduced vertical space in
   compound paragraphs.
div.compound .compound-first, div.compound .compound-middle {
  margin-bottom: 0.5em }

div.compound .compound-last, div.compound .compound-middle {
  margin-top: 0.5em }
*/

div.dedication {
  margin: 2em 5em ;
  text-align: center ;
  font-style: italic }

div.dedication p.topic-title {
  font-weight: bold ;
  font-style: normal }

div.figure {
  margin-left: 2em ;
  margin-right: 2em }

div.footer, div.header {
  clear: both;
  font-size: smaller }

div.line-block {
  display: block ;
  margin-top: 1em ;
  margin-bottom: 1em }

div.line-block div.line-block {
  margin-top: 0 ;
  margin-bottom: 0 ;
  margin-left: 1.5em }

div.sidebar {
  margin: 0 0 0.5em 1em ;
  border: medium outset ;
  padding: 1em ;
  background-color: #ffffee ;
  width: 40% ;
  float: right ;
  clear: right }

div.sidebar p.rubric {
  font-family: sans-serif ;
  font-size: medium }

div.system-messages {
  margin: 5em }

div.system-messages h1 {
  color: red }

div.system-message {
  border: medium outset ;
  padding: 1em }

div.system-message p.system-message-title {
  color: red ;
  font-weight: bold }

div.topic {
  margin: 2em }

h1.section-subtitle, h2.section-subtitle, h3.section-subtitle,
h4.section-subtitle, h5.section-subtitle, h6.section-subtitle {
  margin-top: 0.4em }

h1.title {
  text-align: center }

h2.subtitle {
  text-align: center }

hr.docutils {
  width: 75% }

img.align-left, .figure.align-left, object.align-left {
  clear: left ;
  float: left ;
  margin-right: 1em }

img.align-right, .figure.align-right, object.align-right {
  clear: right ;
  float: right ;
  margin-left: 1em }

img.align-center, .figure.align-center, object.align-center {
  display: block;
  margin-left: auto;
  margin-right: auto;
}

.align-left {
  text-align: left }

.align-center {
  clear: both ;
  text-align: center }

.align-right {
  text-align: right }

/* reset inner alignment in figures */
div.align-right {
  text-align: inherit }

/* div.align-center * { */
/*   text-align: left } */

ol.simple, ul.simple {
  margin-bottom: 1em }

ol.arabic {
  list-style: decimal }

ol.loweralpha {
  list-style: lower-alpha }

ol.upperalpha {
  list-style: upper-alpha }

ol.lowerroman {
  list-style: lower-roman }

ol.upperroman {
  list-style: upper-roman }

p.attribution {
  text-align: right ;
  margin-left: 50% }

p.caption {
  font-style: italic }

p.credits {
  font-style: italic ;
  font-size: smaller }

p.label {
  white-space: nowrap }

p.rubric {
  font-weight: bold ;
  font-size: larger ;
  color: maroon ;
  text-align: center }

p.sidebar-title {
  font-family: sans-serif ;
  font-weight: bold ;
  font-size: larger }

p.sidebar-subtitle {
  font-family: sans-serif ;
  font-weight: bold }

p.topic-title {
  font-weight: bold }

pre.address {
  margin-bottom: 0 ;
  margin-top: 0 ;
  font: inherit }

pre.literal-block, pre.doctest-block, pre.math, pre.code {
  margin-left: 2em ;
  margin-right: 2em }

pre.code .ln { color: grey; } /* line numbers */
pre.code, code { background-color: #eeeeee }
pre.code .comment, code .comment { color: #5C6576 }
pre.code .keyword, code .keyword { color: #3B0D06; font-weight: bold }
pre.code .literal.string, code .literal.string { color: #0C5404 }
pre.code .name.builtin, code .name.builtin { color: #352B84 }
pre.code .deleted, code .deleted { background-color: #DEB0A1}
pre.code .inserted, code .inserted { background-color: #A3D289}

span.classifier {
  font-family: sans-serif ;
  font-style: oblique }

span.classifier-delimiter {
  font-family: sans-serif ;
  font-weight: bold }

span.interpreted {
  font-family: sans-serif }

span.option {
  white-space: nowrap }

span.pre {
  white-space: pre }

span.problematic {
  color: red }

span.section-subtitle {
  /* font-size relative to parent (h1..h6 element) */
  font-size: 80% }

table.citation {
  border-left: solid 1px gray;
  margin-left: 1px }

table.docinfo {
  margin: 2em 4em }

table.docutils {
  margin-top: 0.5em ;
  margin-bottom: 0.5em }

table.footnote {
  border-left: solid 1px black;
  margin-left: 1px }

table.docutils td, table.docutils th,
table.docinfo td, table.docinfo th {
  padding-left: 0.5em ;
  padding-right: 0.5em ;
  vertical-align: top }

table.docutils th.field-name, table.docinfo th.docinfo-name {
  font-weight: bold ;
  text-align: left ;
  white-space: nowrap ;
  padding-left: 0 }

/* "booktabs" style (no vertical lines) */
table.docutils.booktabs {
  border: 0px;
  border-top: 2px solid;
  border-bottom: 2px solid;
  border-collapse: collapse;
}
table.docutils.booktabs * {
  border: 0px;
}
table.docutils.booktabs th {
  border-bottom: thin solid;
  text-align: left;
}

h1 tt.docutils, h2 tt.docutils, h3 tt.docutils,
h4 tt.docutils, h5 tt.docutils, h6 tt.docutils {
  font-size: 100% }

ul.auto-toc {
  list-style-type: none }

</style>
</head>
<body>
<div class="document" id="heatlogo">
<h1 class="title">HeatLogo</h1>
<h2 class="subtitle" id="version-1-0">version: 1.0</h2>

<div class="section" id="general-description">
<h1>General description</h1>
<p>HeatLogo is a tool for generating sequence logos from biological sequence
alignments and for coloring each symbol in the logos according to its P-value.
It can be run on the command line, as a standalone webserver, as a CGI webapp,
or as a python library.</p>
</div>
<div class="section" id="requirement">
<h1>Requirement</h1>
<p>HeatLogo requires the following environment to run:</p>
<ul class="simple">
<li>Python 2.x (x &gt;= 5)</li>
<li>Python array package 'numpy'</li>
<li>Python scientific package 'scipy'</li>
</ul>
<p>('numpy' and 'scipy' can be freely downloaded from: <a class="reference external" href="http://www.scipy.org/Download">http://www.scipy.org/Download</a>)</p>
</div>
<div class="section" id="usage">
<h1>Usage</h1>
<p>Create sequence logos from biological sequence alignments and color symbols
according to P-values:</p>
<pre class="literal-block">
$ heatlogo [options] &lt; sequence_data.fa &gt; sequence_logo.eps
</pre>
</div>
<div class="section" id="options">
<h1>Options</h1>
<blockquote>
<table class="docutils option-list" frame="void" rules="none">
<col class="option" />
<col class="description" />
<tbody valign="top">
<tr><td class="option-group">
<kbd><span class="option">-h</span>, <span class="option">--help</span></kbd></td>
<td>Show all options, their default settings and exit</td></tr>
</tbody>
</table>
</blockquote>
<div class="section" id="input-output-options">
<h2>Input/Output Options</h2>
<blockquote>
<table class="docutils option-list" frame="void" rules="none">
<col class="option" />
<col class="description" />
<tbody valign="top">
<tr><td class="option-group" colspan="2">
<kbd><span class="option">-i</span>, <span class="option">--input <var>FILENAME</var></span></kbd></td>
</tr>
<tr><td>&nbsp;</td><td>Sequence input file (default: stdin)</td></tr>
<tr><td class="option-group" colspan="2">
<kbd><span class="option">-f</span>, <span class="option">--input-format <var>FORMAT</var></span></kbd></td>
</tr>
<tr><td>&nbsp;</td><td>Type of multiple sequence alignment or position weight matrix
file: (clustal, fasta, plain, msf, genbank, nbrf, nexus phylip
stockholm, intelligenetics, table, array, transfac)</td></tr>
<tr><td class="option-group" colspan="2">
<kbd><span class="option">-o</span>, <span class="option">--output <var>FILENAME</var></span></kbd></td>
</tr>
<tr><td>&nbsp;</td><td>Output file (default: stdout)</td></tr>
<tr><td class="option-group" colspan="2">
<kbd><span class="option">-F</span>, <span class="option">--output-format <var>FORMAT</var></span></kbd></td>
</tr>
<tr><td>&nbsp;</td><td><p class="first">Format of output: eps (default), png, png_print, pdf, jpeg, svg, logodata.</p>
<table class="last docutils option-list" frame="void" rules="none">
<col class="option" />
<col class="description" />
<tbody valign="top">
<tr><td class="option-group" colspan="2">
<kbd><span class="option">--pwm-prob <var>FILENAME</var></span></kbd></td>
</tr>
<tr><td>&nbsp;</td><td>Output position weight matrix composed of logarithmic probabilities</td></tr>
<tr><td class="option-group" colspan="2">
<kbd><span class="option">--pwm-pval <var>FILENAME</var></span></kbd></td>
</tr>
<tr><td>&nbsp;</td><td>Output position weight matrix of P-values</td></tr>
</tbody>
</table>
</td></tr>
</tbody>
</table>
</blockquote>
</div>
<div class="section" id="logo-data-options">
<h2>Logo Data Options</h2>
<blockquote>
<table class="docutils option-list" frame="void" rules="none">
<col class="option" />
<col class="description" />
<tbody valign="top">
<tr><td class="option-group" colspan="2">
<kbd><span class="option">-A</span>, <span class="option">--sequence-type <var>TYPE</var></span></kbd></td>
</tr>
<tr><td>&nbsp;</td><td>The type of sequence data: 'protein', 'rna', 'dna', or 'codon'
(See also: --codon-frame)</td></tr>
<tr><td class="option-group" colspan="2">
<kbd><span class="option">-a</span>, <span class="option">--alphabet <var>ALPHABET</var></span></kbd></td>
</tr>
<tr><td>&nbsp;</td><td>The set of symbols to count, e.g. 'AGTC'. All characters not in
the alphabet are ignored. If neither the alphabet nor sequence-type
are specified then heatlogo will examine the input data and make
an educated guess. (See also: --sequence-type, --ignore-lower-case)</td></tr>
<tr><td class="option-group" colspan="2">
<kbd><span class="option">-R</span>, <span class="option">--codon-frame <var>NUMBER</var></span></kbd></td>
</tr>
<tr><td>&nbsp;</td><td>Codon reading frame (default: +1): [+1, +2, +3] indicate the forward
strand frame, and [-1, -2, -3] indicate the reverse strand frame</td></tr>
<tr><td class="option-group" colspan="2">
<kbd><span class="option">-b</span>, <span class="option">--composition <var>COMP</var></span></kbd></td>
</tr>
<tr><td>&nbsp;</td><td><p class="first">The expected composition of the sequences. HeatLogo accepts four
types of formats:</p>
<blockquote class="last">
<ol class="arabic simple">
<li>String: 'auto' (default), 'equiprobable', or 'none'
(do not perform any compositional adjustment, Heatmap: off).
The automatic option uses a typical distribution for
proteins and equiprobable distribution for everything else.</li>
<li>CG percentage.</li>
<li>Explicit distribution of bases, amino acids, or codons
(format: &quot;{'A':10, 'C':40, 'G':40, 'T':10}&quot;).</li>
<li>Tab-delimited file containing each symbol and its expected
composition in each line.</li>
</ol>
</blockquote>
</td></tr>
<tr><td class="option-group" colspan="2">
<kbd><span class="option">-w</span>, <span class="option">--weight <var>NUMBER</var></span></kbd></td>
</tr>
<tr><td>&nbsp;</td><td>The weight of prior data.  Default depends on alphabet length</td></tr>
<tr><td class="option-group" colspan="2">
<kbd><span class="option">-N</span>, <span class="option">--second-data <var>FILENAME</var></span></kbd></td>
</tr>
<tr><td>&nbsp;</td><td>Second dataset file to compare with the input. HeatLogo measures
statistical significance at each column by comparing the number of
symbols appered in input and second datasets.
Accepted formats: clustal, fasta, plain, msf, genbank, nbrf, nexus,
phylip, stockholm, intelligenetics, table, array.</td></tr>
<tr><td class="option-group" colspan="2">
<kbd><span class="option">-B</span>, <span class="option">--second-composition <var>COMP</var></span></kbd></td>
</tr>
<tr><td>&nbsp;</td><td>The expected composition of the second sample sequences. The supported
formats are the same as those of '--composition'.</td></tr>
<tr><td class="option-group" colspan="2">
<kbd><span class="option">-W</span>, <span class="option">--second-weight <var>NUMBER</var></span></kbd></td>
</tr>
<tr><td>&nbsp;</td><td>The prior weight for the second dataset. Default depends on alphabet length.</td></tr>
<tr><td class="option-group" colspan="2">
<kbd><span class="option">-U</span>, <span class="option">--units <var>NUMBER</var></span></kbd></td>
</tr>
<tr><td>&nbsp;</td><td>A unit of entropy ('bits' (default), 'nats', 'digits'), or a unit of free
energy ('kT', 'kJ/mol', 'kcal/mol'), or 'probability' for probabilities</td></tr>
</tbody>
</table>
</blockquote>
</div>
<div class="section" id="heat-map-options">
<h2>Heat Map Options</h2>
<p>These options affect the format of the sequence logo in the heat map
mode (See also Color Options).</p>
<blockquote>
<table class="docutils option-list" frame="void" rules="none">
<col class="option" />
<col class="description" />
<tbody valign="top">
<tr><td class="option-group" colspan="2">
<kbd><span class="option">-S</span>, <span class="option">--stats-test <var>TEST</var></span></kbd></td>
</tr>
<tr><td>&nbsp;</td><td>Specify a statistical test to calculate p-values: 'binomial', 't-test',
'z-score' (default: t-test)</td></tr>
<tr><td class="option-group" colspan="2">
<kbd><span class="option">-H</span>, <span class="option">--heat-scheme <var>SCHEME</var></span></kbd></td>
</tr>
<tr><td>&nbsp;</td><td><dl class="first docutils">
<dt>Specify a heatmap color scheme: 'magenta-cyan', 'red-blue', 'red-green',</dt>
<dd>'thermography' (See also: '--p-color').</dd>
<dt>--p-color COLOR PVALUE   Specify a color for a p-value (p-value range: -1 &lt; p &lt; 1).</dt>
<dd>(e.g. --p-color blue 0.5 --p-color red 0.25)</dd>
</dl>
<table class="last docutils option-list" frame="void" rules="none">
<col class="option" />
<col class="description" />
<tbody valign="top">
<tr><td class="option-group" colspan="2">
<kbd><span class="option">--hide-colorkey</span></kbd></td>
</tr>
<tr><td>&nbsp;</td><td>Toggle switch to hide a heatmap color key on the right side of a
sequence logo (default: off)</td></tr>
</tbody>
</table>
</td></tr>
</tbody>
</table>
</blockquote>
</div>
<div class="section" id="color-options">
<h2>Color Options</h2>
<p>Colors can be specified using CSS2 syntax. e.g. 'red', '#FF0000', etc
(See also Heat Map Options).</p>
<blockquote>
<table class="docutils option-list" frame="void" rules="none">
<col class="option" />
<col class="description" />
<tbody valign="top">
<tr><td class="option-group" colspan="2">
<kbd><span class="option">-C</span>, <span class="option">--color-scheme <var>SCHEME</var></span></kbd></td>
</tr>
<tr><td>&nbsp;</td><td><dl class="first docutils">
<dt>Specify a standard color scheme (auto, base pairing, charge, chemistry,</dt>
<dd>classic, dna codon, hydrophobicity, monochrome, rna codon; Heatmap: off)</dd>
<dt>--c-color COLOR SYMBOLS DESCRIPTION</dt>
<dd><dl class="first last docutils">
<dt>Specify symbol colors (e.g. --c-color black AG 'Purine'</dt>
<dd>--c-color red TC 'Pyrimidine'; Heatmap: off)</dd>
</dl>
</dd>
</dl>
<table class="last docutils option-list" frame="void" rules="none">
<col class="option" />
<col class="description" />
<tbody valign="top">
<tr><td class="option-group" colspan="2">
<kbd><span class="option">--default-color <var>COLOR</var></span></kbd></td>
</tr>
<tr><td>&nbsp;</td><td>Symbol color if not otherwise specified.</td></tr>
</tbody>
</table>
</td></tr>
</tbody>
</table>
</blockquote>
</div>
<div class="section" id="transformations">
<h2>Transformations</h2>
<p>Optional transformations of the sequence data.</p>
<blockquote>
<table class="docutils option-list" frame="void" rules="none">
<col class="option" />
<col class="description" />
<tbody valign="top">
<tr><td class="option-group" colspan="2">
<kbd><span class="option">--ignore-lower-case</span></kbd></td>
</tr>
<tr><td>&nbsp;</td><td>Disregard lower case letters and only count upper case letters in sequences.</td></tr>
</tbody>
</table>
<p class="attribution">&mdash;reverse                Simply reverse input sequences.</p>
</blockquote>
<blockquote>
<table class="docutils option-list" frame="void" rules="none">
<col class="option" />
<col class="description" />
<tbody valign="top">
<tr><td class="option-group">
<kbd><span class="option">--complement</span></kbd></td>
<td>Complement DNA sequences.</td></tr>
</tbody>
</table>
</blockquote>
</div>
<div class="section" id="logo-format-options">
<h2>Logo Format Options</h2>
<p>These options control the format and display of the sequence logo.</p>
<blockquote>
<table class="docutils option-list" frame="void" rules="none">
<col class="option" />
<col class="description" />
<tbody valign="top">
<tr><td class="option-group" colspan="2">
<kbd><span class="option">-I</span>, <span class="option">--first-index <var>INDEX</var></span></kbd></td>
</tr>
<tr><td>&nbsp;</td><td>Index of first position in sequence data (default: 1)</td></tr>
<tr><td class="option-group" colspan="2">
<kbd><span class="option">-s</span>, <span class="option">--start <var>INDEX</var></span></kbd></td>
</tr>
<tr><td>&nbsp;</td><td>Lower bound (i.e. start position) of sequence to display</td></tr>
<tr><td class="option-group" colspan="2">
<kbd><span class="option">-e</span>, <span class="option">--end <var>INDEX</var></span></kbd></td>
</tr>
<tr><td>&nbsp;</td><td>Upper bound (i.e. end position) of sequence to display</td></tr>
<tr><td class="option-group" colspan="2">
<kbd><span class="option">-L</span>, <span class="option">--size <var>LOGOSIZE</var></span></kbd></td>
</tr>
<tr><td>&nbsp;</td><td>Specify a standard logo size (small, medium (default), large)</td></tr>
<tr><td class="option-group" colspan="2">
<kbd><span class="option">-n</span>, <span class="option">--stacks-per-line <var>COUNT</var></span></kbd></td>
</tr>
<tr><td>&nbsp;</td><td>Maximum number of logo stacks per logo line (default: 40)</td></tr>
<tr><td class="option-group" colspan="2">
<kbd><span class="option">-t</span>, <span class="option">--title <var>TEXT</var></span></kbd></td>
</tr>
<tr><td>&nbsp;</td><td>Logo title text.</td></tr>
<tr><td class="option-group" colspan="2">
<kbd><span class="option">-l</span>, <span class="option">--label <var>TEXT</var></span></kbd></td>
</tr>
<tr><td>&nbsp;</td><td>A figure label (e.g. '2a').</td></tr>
</tbody>
</table>
<p>-X, --show-xaxis YES/NO      Display sequence numbers along x-axis? (default: True)</p>
<table class="docutils option-list" frame="void" rules="none">
<col class="option" />
<col class="description" />
<tbody valign="top">
<tr><td class="option-group" colspan="2">
<kbd><span class="option">-x</span>, <span class="option">--xlabel <var>TEXT</var></span></kbd></td>
</tr>
<tr><td>&nbsp;</td><td><p class="first">X-axis label</p>
<table class="last docutils option-list" frame="void" rules="none">
<col class="option" />
<col class="description" />
<tbody valign="top">
<tr><td class="option-group" colspan="2">
<kbd><span class="option">--annotate <var>TEXT</var></span></kbd></td>
</tr>
<tr><td>&nbsp;</td><td>A comma separated list of custom stack annotations (e.g. '1,3,4,5,6,7').
Annotation list must be same length as sequences.</td></tr>
</tbody>
</table>
</td></tr>
<tr><td class="option-group" colspan="2">
<kbd><span class="option">-M</span>, <span class="option">--yaxis <var>UNIT</var></span></kbd></td>
</tr>
<tr><td>&nbsp;</td><td>Height of yaxis in units. (Default: Maximum value with uninformative prior.)</td></tr>
</tbody>
</table>
<p>-Y, --show-yaxis YES/NO      Display entropy scale along y-axis? (default: True)</p>
<table class="docutils option-list" frame="void" rules="none">
<col class="option" />
<col class="description" />
<tbody valign="top">
<tr><td class="option-group" colspan="2">
<kbd><span class="option">-y</span>, <span class="option">--ylabel <var>TEXT</var></span></kbd></td>
</tr>
<tr><td>&nbsp;</td><td>Y-axis label (default depends on plot type and units)</td></tr>
</tbody>
</table>
<p>-E, --show-ends YES/NO       Label the ends of the sequence? (default: False)</p>
<table class="docutils option-list" frame="void" rules="none">
<col class="option" />
<col class="description" />
<tbody valign="top">
<tr><td class="option-group" colspan="2">
<kbd><span class="option">-P</span>, <span class="option">--fineprint <var>TEXT</var></span></kbd></td>
</tr>
<tr><td>&nbsp;</td><td><p class="first">The fine print (default: heatlogo version)</p>
<table class="docutils option-list" frame="void" rules="none">
<col class="option" />
<col class="description" />
<tbody valign="top">
<tr><td class="option-group" colspan="2">
<kbd><span class="option">--ticmarks <var>NUMBER</var></span></kbd></td>
</tr>
<tr><td>&nbsp;</td><td>Distance between ticmarks (default: 1.0)</td></tr>
</tbody>
</table>
<p>--errorbars YES/NO       Display error bars? (default: True)</p>
<p class="last">--reverse-stacks YES/NO  Draw stacks with largest letters on top? (default: True)</p>
</td></tr>
</tbody>
</table>
</blockquote>
</div>
<div class="section" id="advanced-format-options">
<h2>Advanced Format Options</h2>
<p>These options provide fine control over the display of the sequence logo.</p>
<blockquote>
<table class="docutils option-list" frame="void" rules="none">
<col class="option" />
<col class="description" />
<tbody valign="top">
<tr><td class="option-group" colspan="2">
<kbd><span class="option">-k</span>, <span class="option">--stack-width <var>POINTS</var></span></kbd></td>
</tr>
<tr><td>&nbsp;</td><td><p class="first">Width of a logo stack (default: 10.8)</p>
<table class="docutils option-list" frame="void" rules="none">
<col class="option" />
<col class="description" />
<tbody valign="top">
<tr><td class="option-group" colspan="2">
<kbd><span class="option">--aspect-ratio <var>POINTS</var></span></kbd></td>
</tr>
<tr><td>&nbsp;</td><td>Ratio of stack height to width (default: 5)</td></tr>
</tbody>
</table>
<p>--box YES/NO             Draw boxes around symbols? (default: no)</p>
<table class="docutils option-list" frame="void" rules="none">
<col class="option" />
<col class="description" />
<tbody valign="top">
<tr><td class="option-group" colspan="2">
<kbd><span class="option">--resolution <var>DPI</var></span></kbd></td>
</tr>
<tr><td>&nbsp;</td><td>Bitmap resolution in dots per inch (DPI) (Default: 96 DPI, except
png_print, 600 DPI). Low resolution bitmaps (DPI&lt;300) are antialiased.</td></tr>
</tbody>
</table>
<dl class="last docutils">
<dt>--scale-width YES/NO     Scale the visible stack width by the fraction of symbols in the column?</dt>
<dd>(i.e. columns with many gaps of unknowns are narrow.) (Default: yes)</dd>
</dl>
</td></tr>
</tbody>
</table>
</blockquote>
</div>
<div class="section" id="heatlogo-server">
<h2>HeatLogo Server</h2>
<p>Run a standalone webserver on a local port.</p>
<blockquote>
<table class="docutils option-list" frame="void" rules="none">
<col class="option" />
<col class="description" />
<tbody valign="top">
<tr><td class="option-group">
<kbd><span class="option">--serve</span></kbd></td>
<td>Start a standalone HeatLogo server for creating sequence logos.</td></tr>
</tbody>
</table>
<p class="attribution">&mdash;port PORT              Listen to this local port. (Default: 8080)</p>
</blockquote>
</div>
</div>
<div class="section" id="examples">
<h1>Examples</h1>
<p>For help on the command line interface, type:</p>
<pre class="literal-block">
$ ./heatlogo --help
</pre>
<p>To build a simple logo, execute:</p>
<pre class="literal-block">
$ ./heatlogo &lt; seqs.fa &gt; logo.eps
</pre>
<p>If the input data is codon, type:</p>
<pre class="literal-block">
$ ./heatlogo --sequence-type 'codon' &lt; seqs.fa &gt; logo.eps
</pre>
<p>HeatLogo takes into account expected composition of sequences using the following commands.</p>
<ul>
<li><p class="first">String (for DNA, RNA, protein, and codon):</p>
<pre class="literal-block">
$ ./heatlogo --composition ['auto'| 'equiprobable'|'none'] &lt; seqs.fa &gt; logo.eps
</pre>
</li>
<li><p class="first">Explicit distribution of symbols (for DNA, RNA, protein, and codon):</p>
<pre class="literal-block">
$ ./heatlogo --composition &quot;{'A':30, 'C':20, 'G':20, 'T':30}&quot; &lt; seqs.fa &gt; logo.eps
</pre>
</li>
<li><p class="first">CG percentage (only applicable for DNA or RNA):</p>
<pre class="literal-block">
$ ./heatlogo --composition 40% &lt; seqs.fa &gt; logo.eps
</pre>
</li>
<li><p class="first">Tab-delimited file containing each symbol and its composition (for DNA, RNA, protein, and codon):</p>
<pre class="literal-block">
$ ./heatlogo --composition COMPOSITON_FILENAME &lt; seqs.fa &gt; logo.eps
</pre>
<ul>
<li><p class="first">Note: tab-delimited file format</p>
<table border="1" class="docutils">
<colgroup>
<col width="29%" />
<col width="31%" />
<col width="40%" />
</colgroup>
<thead valign="bottom">
<tr><th class="head">DNA or RNA</th>
<th class="head">protein</th>
<th class="head">codon</th>
</tr>
</thead>
<tbody valign="top">
<tr><td>A   24.3</td>
<td>A   13.8</td>
<td>AAA   0.080</td>
</tr>
<tr><td>T   24.3</td>
<td>C   12.1</td>
<td>AAC   0.018</td>
</tr>
<tr><td>G   25.7</td>
<td>D    3.4</td>
<td>AAG   0.409</td>
</tr>
<tr><td>C   25.7</td>
<td>E    8.9 ...</td>
<td>AAT   0.004 ...</td>
</tr>
<tr><td>(4 lines)</td>
<td>(20 lines)</td>
<td>(64 lines)</td>
</tr>
</tbody>
</table>
</li>
</ul>
</li>
</ul>
<p>For calculating statistical significance of each amino acid,
specify a statistical test as follows:</p>
<pre class="literal-block">
$ ./heatlogo --stats-test 'binomial' &lt; seqs.fa &gt; logo.eps
</pre>
<p>If there is negative or paired dataset against the input, HeatLogo computes P-values
by comparing each column of the input with the column of the negative or paired dataset:</p>
<pre class="literal-block">
$ ./heatlogo --second-data FILENAME &lt; seqs.fa &gt; logo.eps
</pre>
<p>Heatmap schemes implemented in HeatLogo can be changed with the following option:</p>
<pre class="literal-block">
$ ./heatlogo --heat-scheme 'red-blue' &lt; seq.fa &gt; logo.eps
</pre>
<p>To generate position weight matrices of P-values and log-odds ratios, type:</p>
<pre class="literal-block">
$ ./heatlogo --pwm-pval FILENAME --pwm-prob FILENAME &lt; seqs.fa &gt; logo.eps
</pre>
<p>HeatLogo also generates sequence logos with usual color schemes by turning off
the heatmap mode:</p>
<pre class="literal-block">
$ ./heatlogo --color-scheme 'base pairing' &lt; seqs.fa &gt; logo.eps
</pre>
</div>
<div class="section" id="color-schemes">
<h1>Color Schemes</h1>
<p>Color schemes (i.e. color schemes which are not heatmap mode) of usual
sequence logos are listed as follows.</p>
<div class="section" id="amino-acids">
<h2>Amino Acids</h2>
<p>All color schemes for amino acids are based on their physicochemical properties.</p>
<ul class="simple">
<li>Hydrophobicity</li>
</ul>
<table border="1" class="docutils">
<colgroup>
<col width="41%" />
<col width="41%" />
<col width="17%" />
</colgroup>
<thead valign="bottom">
<tr><th class="head">Physicochemical prop.</th>
<th class="head">Amino acid</th>
<th class="head">Colors</th>
</tr>
</thead>
<tbody valign="top">
<tr><td>Hydrophilic</td>
<td>R, K, D, E, N, Q</td>
<td>blue</td>
</tr>
<tr><td>Neutral</td>
<td>S, G, H, T, A, P</td>
<td>green</td>
</tr>
<tr><td>Hydrophobic</td>
<td>Y, V, M, C, L, F, I, W</td>
<td>black</td>
</tr>
</tbody>
</table>
<ul class="simple">
<li>Chemistry</li>
</ul>
<table border="1" class="docutils">
<colgroup>
<col width="41%" />
<col width="41%" />
<col width="17%" />
</colgroup>
<thead valign="bottom">
<tr><th class="head">Physicochemical prop.</th>
<th class="head">Amino acid</th>
<th class="head">Colors</th>
</tr>
</thead>
<tbody valign="top">
<tr><td>Polar</td>
<td>G, S, T, Y, C</td>
<td>green</td>
</tr>
<tr><td>Neutral</td>
<td>N, Q</td>
<td>purple</td>
</tr>
<tr><td>Basic</td>
<td>K, R, H</td>
<td>blue</td>
</tr>
<tr><td>Acidic</td>
<td>D, E</td>
<td>red</td>
</tr>
<tr><td>Hydrophobic</td>
<td>P, A, W, F, L, I, M, V</td>
<td>black</td>
</tr>
</tbody>
</table>
<ul class="simple">
<li>Charge</li>
</ul>
<table border="1" class="docutils">
<colgroup>
<col width="50%" />
<col width="29%" />
<col width="21%" />
</colgroup>
<thead valign="bottom">
<tr><th class="head">Physicochemical prop.</th>
<th class="head">Amino acid</th>
<th class="head">Colors</th>
</tr>
</thead>
<tbody valign="top">
<tr><td>Positive</td>
<td>K, R, H</td>
<td>blue</td>
</tr>
<tr><td>Negative</td>
<td>D, E</td>
<td>red</td>
</tr>
</tbody>
</table>
</div>
<div class="section" id="codons">
<h2>Codons</h2>
<p>The color scheme is based on physicochemical properties of amino acids
encoded by codons according to the standard genetic code table.</p>
<table border="1" class="docutils">
<colgroup>
<col width="37%" />
<col width="22%" />
<col width="42%" />
</colgroup>
<thead valign="bottom">
<tr><th class="head">Physicochemical prop.</th>
<th class="head">Amino acid</th>
<th class="head">Colors</th>
</tr>
</thead>
<tbody valign="top">
<tr><td>Polar positive</td>
<td>H, K, R</td>
<td>light blue</td>
</tr>
<tr><td>Polar negative</td>
<td>D, E</td>
<td>red</td>
</tr>
<tr><td>Polar neutral</td>
<td>S, T, N, Q</td>
<td>green</td>
</tr>
<tr><td>Non-polar aliphatic</td>
<td>A, V, L, I</td>
<td>blue</td>
</tr>
<tr><td>Non-polar aromatic</td>
<td>F, W, Y</td>
<td>magenta</td>
</tr>
<tr><td>Special residue</td>
<td>P, G, C</td>
<td>brown (P, G), yellow (C)</td>
</tr>
<tr><td>Stop codons</td>
<td><tt class="docutils literal">*</tt></td>
<td>black</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
</body>
</html>
